### PR TITLE
Implement the components required for incentive promotion

### LINKED ISSUE #51453: [WC Payments NOX] Integrate WooPayments incentives UI
### Description

If there is an active WooPayments incentive, surface it on the Payments settings page.

### Acceptance criteria

* The action incentive UI is present when there is an active incentive.
* The switch incentive UI is present when there is an active incentive. 

### Designs
Design post: pfHfG4-q7-p2
Figma: xO8YdpKfaVoP2d8NvweZ5I-fi-1428_21437

### Dev notes
Reactifying the Payments settings page: pe4Hk0-1ly-p2
Populate react-powered main payments settings screen PR: https://github.com/woocommerce/woocommerce/pull/50825

### Additional context
PRD: 1wMfX8CAjDnOp24tcU-PADB0pOt3cz7mGdyZYeRBxx2Y-gdoc

cc @timmy5685

### PR DESCRIPTION
**Pull request by dmallory42**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Partially closes #51453  .

Noting that this PR addresses the creation of the components, and will require a separate PR addressing the logic to pull and check for incentives.

### Screenshots

Link to designs: xO8YdpKfaVoP2d8NvweZ5I-fi-3106_71830

Incentive Modal:

Screenshot 2024-11-08 at 10 41 43

 
**Note** As you can see, it looks a little off what we have in the design. We have a limitation here which is that the `<Modal>` component has the close button as part of a separate "top header" section, which makes it look a little but weird. I can think of two possible solutions:

1. Remove the top header completely, and instead of the Close icon in the top right, implement a close button at the bottom next to the `Get started` button
2. Remove the top header, and implement our own custom close icon in the top right corner. I don't really like this solution, because it feels hacky, and this is something that should be addressed on the top level `Modal` component (which I don't think is in scope of the project)

cc. @orcunomattic - do you have a preference on how to address this issue?

Incentive Banner and highlighted recommendation:

Screenshot 2024-11-08 at 10 43 52

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Copy the following code into the `settings-payments-main.tsx` file, replacing the existing content.

```js
/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import './settings-payments-main.scss';
import { PaymentGateways } from '~/settings-payments/components/payment-gateways';
import { OtherPaymentGateways } from '~/settings-payments/components/other-payment-gateways';
import { IncentiveBanner } from '~/settings-payments/components/incentive-banner';
import { IncentiveModal } from '~/settings-payments/components/incentive-modal';

export const SettingsPaymentsMain = () => {
	return (
		<>
			<div className="settings-payments-main__container">
				<IncentiveModal isOpen={ true } onClose={ () => {} } onSubmit={ () => {} } />
				<IncentiveBanner />
				<PaymentGateways />
				<OtherPaymentGateways />
			</div>
		</>
	);
};

export default SettingsPaymentsMain;

```

2. Run the command `pnpm run --filter=@woocommerce/plugin-woocommerce watch:build `
3. Go to the `WooCommerce > Settings > Payments` page. You should see the incentives modal. It won't close, or do anything currently.
4. Remove the `<IncentiveModal isOpen={ true } onClose={ () => {} } onSubmit={ () => {} } />` line from the code above and wait for it to recompile. 
5. Refresh the page, you should see the Incentive Banner, and the WooPayments recommendation should be highlighted.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Pill is an unused import.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Since we aren't expecting its CSS classes from a variable, we could directly use a literal here.

```suggestion
		<Card className="woocommerce-incentive-banner" isRounded={ true }>
```

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: The same suggestion applies here, though it's understandable if you'd like to use the curly braces for consistency with other custom props. For all `className` instances, I would still recommend using the double-quotes.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]:
```suggestion
							'Use the native payments solution built and supported by Woo to accept online and in-person payments, track revenue, and handle all payment activity in one place.',
```

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx]: Another unused import.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx]:
```suggestion
										'Use the native payments solution built and supported by Woo to accept online and in-person payments, track revenue, and handle all payment activity in one place.',
```

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx]: Wouldn't this keep the button in not-busy state before submission and always-busy after submission?

File: plugins/woocommerce/client/admin/client/settings-payments/components/offline-payment-gateways/offline-payment-gateways.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/offline-payment-gateways/offline-payment-gateways.tsx]: Thanks for moving all component files in their respective folders!

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx]: Alternatively: (too nitpicky; use whatever suits your style)

```suggestion
					className: `woocommerce-list__item--payment-gateway ${ hasIncentive ?? 'has-incentive' }`
```

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx]: Question: Why is this property called `method_description` (and not just description)?

File: plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx:29]: Maybe add a comment here that `message` will override `status`?

File: plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx:28]: You don't seem to be using `info` anywhere.

**Review by anu-rock — COMMENTED** [review]
Code-only review. Cannot wait to take it for a spin (manual testing). I'll watch out for testing instructions.

**Comment by anu-rock** [comment]
@dmallory42 In case you haven't gotten to it yet, a StatusBadge test is failing: https://github.com/woocommerce/woocommerce/actions/runs/11740607410/job/32707528040?pr=52453.

**Comment by anu-rock** [comment]
> Remove the top header, and implement our own custom close icon in the top right corner. I don't really like this solution, because it feels hacky, and this is something that should be addressed on the top level Modal component (which I don't think is in scope of the project)

Is it possible to add CSS overrides to layer the header (including the close button) on top of the modal content?

**Comment by anu-rock** [comment]
I just finished manually testing it. Also, double-checking whether the WooPayments entry in the Payment Providers section will be highlighted only if the incentive banner is dismissed?
___

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| Replace the file contents for `settings-payments-main.tsx` with the one provided | ✅ Switch incentive modal popped up on page load and looked the same as in the screenshot. |  |
| Remove the modal component in `settings-payments-main.tsx` | ✅ Action incentive banner, earlier in the background, was clearly visible. It matched the Figma design. |  |
| Tested the modal in mobile view (Chrome/iPhone 14 Pro Max) | ❌ The modal content was not fully visible and it introduced a horizontal scroll. There was a lot of empty space, leaving less space for the text. | screenshot of incentive modal in mobile view |
| Tested the banner in mobile view (Chrome/iPhone 14 Pro Max) | ❌ The left image was cut off. There was a lot of empty space, leaving less space for the text. | screenshot of incentive banner in mobile view |

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (chihsuan) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: I know we have a lot of code that use SVG in JS files 😅, but it seems that we should only use it when we need to set its props dynamically (pe4Hk0-1d6-p2#comment-1459).

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx]: I actually don't know, but `method_description` and `description` are two _separate_ properties on the gateway object, and in this instance `method_description` is the one which returns what we would like to show on the frontend. Probably there's some historical context there for why it is like that.

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.tsx]: I like the suggestion! I've applied it.

File: plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx:28]: ah, I think I had an `info` status in a previous iteration of this, but then opted not to use it.

File: plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.tsx:29]: Good idea. Thanks!

**Comment by dmallory42** [comment]
Thanks for the code reveiw, @anu-rock! I've addressed your suggestions. With regards to the mobile view, I think probably it'll be worth addressing in a separate PR because there will be other design fixes that will be needed prior to project completion.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Interesting, I wasn't aware of that so thanks for sharing! Sorry if I missed it in the PR link, but is there an example of the correct way to use SVG-in-JS somewhere in the Core codebase already?

**Comment by dmallory42** [comment]
> Is it possible to add CSS overrides to layer the header (including the close button) on top of the modal content?

Maybe, that's a good idea. I can try and play with it and see if that helps.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.scss
Comment (oaratovskyi) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.scss]: I think we don't need this comments since variable values may change

File: plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.scss
Comment (oaratovskyi) [inline plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateways/payment-gateways.scss:7]: let's address this in the PR where we'll fix mobile CSS

**Review by oaratovskyi — APPROVED** [review]
@dmallory42 looks really good! 
I've fixed lint error and now it's green (except changelog). There are a few things in this PR that could be improved but I'll leave it up to you whether to fix it now or in a follow-up PR, thanks!

I noticed that incentive banner has type size different from the design and paddings too

image

image

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.scss
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.scss]: Agreed. Removed!

**Comment by dmallory42** [comment]
Thanks @oaratovskyi, I've made some changes to address your comments and make the banner design closer to the design (it's not 100%, but we can fix anything outstanding before shipping).

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (chihsuan) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Yeah, we have a lot of examples of that. It's actually not SVG-in-JS, but normal asset files.

You can put the SVG file in the `plugins/woocommerce/assets/images` folder and then use it like this:

https://github.com/woocommerce/woocommerce/blob/15a3ca848d0f28c9dffc26a2374734db129003e1/plugins/woocommerce/client/admin/client/task-lists/setup-task-list/components/task-headers/customize-store.tsx#L35-L38

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (chihsuan) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Could you update the code to use `createInterpolateElement` from `@wordpress/element` instead of `interpolateComponents`? Thanks!

For context: 
- https://github.com/woocommerce/woocommerce/issues/37312

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx
Comment (chihsuan) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx:102]: For privacy reasons, can we add `rel="noreferrer"` to the external link? 🙂

Comment (chihsuan) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx:102]: Also we should added `type="external"` so it can open in a new tab (only `target="_blank"` doesn't work).

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (chihsuan) [inline plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx:67]: Same here for the external link.

**Review by chihsuan — COMMENTED** [review]
Nice work on the this! Just left a few suggestions but feel free to address them in a follow up.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Ah, that's perfect - I'll update accordingly 🙂  thanks for the pointers!

**Review by anu-rock — APPROVED** [review]
LGTM after the latest changes. Thanks for including my feedback! There are a few outstanding comments from @chihsuan, but other than that looks good.

**Comment by anu-rock** [comment]
> With regards to the mobile view, I think probably it'll be worth addressing in a separate PR because there will be other design fixes that will be needed prior to project completion.

Cool! Could you create a separate issue for fixing design/mobile issues? I'll dump my findings to date directly there.

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-banner/incentive-banner.tsx]: Thanks for pointing it out, handled in https://github.com/woocommerce/woocommerce/pull/52453/commits/12b2fec17da6c7563c3b8dd4a7ad8e17ea0e16b0!

File: plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx
Comment (dmallory42) [reply plugins/woocommerce/client/admin/client/settings-payments/components/incentive-modal/incentive-modal.tsx:102]: Ah, I was wondering why it wasn't opening in a new tab with `target="_blank"`. Thanks!

**Comment by dmallory42** [comment]
@anu-rock Thanks! Please add them to https://github.com/woocommerce/woocommerce/issues/52732

**Pull request merged**
